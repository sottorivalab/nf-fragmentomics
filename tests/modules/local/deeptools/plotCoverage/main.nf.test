nextflow_process {

    name "Test Process PLOTCOVERAGE"
    script "modules/local/deeptools/plotCoverage/main.nf"
    process "PLOTCOVERAGE"

    test("Should run PLOTCOVERAGE without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                
                """
                input[0] = [
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
