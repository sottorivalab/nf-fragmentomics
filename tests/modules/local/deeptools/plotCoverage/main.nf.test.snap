{
    "Should run PLOTCOVERAGE without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "caseid": "test",
                            "sampleid": "test",
                            "timepoint": "T1"
                        },
                        "test.paired_end.sorted_coverage.png:md5,675ba6a3a5b0bab697bf11d266f61c06",
                        "test.paired_end.sorted_coverage.tab:md5,8a910e49beef5599b3cbeee22d06bffd"
                    ]
                ],
                "bamcoverage": [
                    [
                        {
                            "caseid": "test",
                            "sampleid": "test",
                            "timepoint": "T1"
                        },
                        "test.paired_end.sorted_coverage.png:md5,675ba6a3a5b0bab697bf11d266f61c06",
                        "test.paired_end.sorted_coverage.tab:md5,8a910e49beef5599b3cbeee22d06bffd"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-15T11:08:22.389654"
    }
}